static/conformance-view.js

```diff
@@ -0,0 +1,314 @@
+            allowAdd: true
+        };
+
+        this.container.on('destroy', function () {
+            this.eventHub.emit("conformanceViewClose", this.editorId);
+            this.eventHub.unsubscribe();
+        }, this);
+
+        this.container.on('open', function () {
+            this.eventHub.emit("conformanceViewOpen", this.editorId);
+        }, this);
+
+        this.eventHub.on('editorChange', this.onEditorChange, this);
+        this.eventHub.on('editorClose', this.onEditorClose, this);
+
+        this.addCompilerButton.on('click', _.bind(function() {
+            this.addCompilerSelector();
+            this.saveState();
+        }, this));
+
+        this.compileButton.on('click', _.bind(function() {
+            this.selectorList.find('.status').css("visibility", "hidden");
+            this.compileAll();
+        }, this));
+
+        if (state.compilers) {
+            _.each(state.compilers, _.bind(function (config) {
+                config.cv = this.nextSelectorId;
+                this.nextSelectorId++;
+                this.addCompilerSelector(config);
+            }, this));
+        }
+
+        this.handleToolbarUI();
+    }
+
+    Conformance.prototype.setTitle = function (compilerCount) {
+        this.container.setTitle("Conformance viewer (Editor #" + this.editorId + ") " + (
+                compilerCount !== 0 ? (compilerCount + "/" + this.maxCompilations) : ""
+            ));
+    };
+
+    Conformance.prototype.addCompilerSelector = function(config) {
+        if (!config) {
+            config = {
+                // Have we been compiled? If so, what is the result?
+                status: {
+                    // 0: Not compiled; 1: Ok; 2: Warnings; 3: Error
+                    code: 0,
+                    // If compiled, what the compiled output to stdout/err
+                    text: ""
+                },
+                // Code we have
+                cv: this.nextSelectorId,
+                // Compiler id which is being used
+                compilerId: "",
+                // Options which are in use
+                options: "",
+            };
+            this.nextSelectorId++;
+        }
+        
+        config.status.code = Number(config.status.code);
+        config.cv = Number(config.cv);
+
+        var newSelector = $("<select>")
+            .attr("class", "compiler-picker")
+            .attr("placeholder", "Select a compiler...")
+            .attr("data-cv", config.cv)
+            .on('change', _.bind(function() {
+                this.saveState();
+            }, this));
+
+        var status = $("<span>");
+
+        var compilationOptions = $("<input>")
+            .attr("class", "options form-control")
+            .attr("type", "text")
+            .attr("size", "256")
+            .attr("placeholder", "Compiler options...")
+            .attr("data-cv", config.cv)
+            .val(config.options);
+
+        this.selectorList.append($("<tr>")
```

Comment:
Would be nice if possible to use a template defined in the `pug` file and then modify it. This is totally fine but I've tried my best to not put markup in the code where possible. I'd be happy to get this checked in then if you don't have time or need a hand I can do a PR to show what I mean by this :)